src/Npgsql/Internal/PgWriter.cs

```diff
@@ -271,13 +271,15 @@ public void WriteDouble(double value)
         Advance(sizeof(double));
     }
 
-    public void WriteChars(ReadOnlySpan<char> data, Encoding encoding)
+    public void WriteChars(ReadOnlySpan<char> data, Encoding encoding, int? encodedByteLength = null)
```

Comment:
I don't think we should make these error-prone methods public. Instead we can apply the optimization to the internal converters. I'm fine having these signatures accessible internally.